dotnet-authserver/src/TeacherIdentity.AuthServer/Journeys/TrnTokenSignInJourney.cs

```diff
@@ -10,11 +14,29 @@ public class TrnTokenSignInJourney : SignInJourney
     {
     }
 
+    public override async Task<IActionResult> CreateUser(string currentStep)
+    {
+        try
+        {
+            var user = await CreateUserHelper.CreateUserWithTrn(AuthenticationState);
+
+            AuthenticationState.OnUserRegistered(user);
+            await AuthenticationState.SignIn(HttpContext);
+        }
+        catch (UniqueConstraintException ex) when (ex.IsUniqueIndexViolation(User.TrnUniqueIndexName))
+        {
+            // We currently do not handle trn index violations for the trn token sign in journey
+        }
```

Comment:
Until we do handle this may be better to let the exception bubble up
```suggestion
            throw;
        }
```